lib/matplotlib/patches.py

```diff
@@ -1066,13 +1066,21 @@ def set_xy(self, xy):
         ----------
         xy : (N, 2) array-like
             The coordinates of the vertices.
+
+        Notes
+        -----
+        Unlike `~.path.Path`, we do not ignore the last input vertex. If the
+        polygon is meant to be closed, and the last point of the polygon is not
+        equal to the first, we assume that the user has not explicitly passed a
+        ``CLOSEPOLY`` vertex, and add it ourselves.
         """
         xy = np.asarray(xy)
+        N, _ = xy.shape
         if self._closed:
-            if len(xy) and (xy[0] != xy[-1]).any():
+            if N == 1 or N > 1 and (xy[0] != xy[-1]).any():
```

Comment:
Could you add a quick comment about the `if` statements you've changed just to say what they're doing? And maybe `nverts` is a better variable name than just `N`?

Reply:
I was reticent to add these comments, because technically they are very brittle checks as-written, since we don't guarantee anything about the coordinates of a CLOSEPOLY vertex, but this code uses the assumption that these vertices will "correctly" match the start point of the curve.

In particular, both Python (e.g. `Path.iter_bezier`) and C++ code (i.e. handlers for `agg:path_flags_close`) actively ignore this vertex's value, because it's known to hold "trash" sometimes, especially after cleaning NaN arrays, etc (weird stuff like #17885 and #17914).

But, I added comments that make it more clear what the if statements are *nominally* doing, since you think it's a good idea. I personally like `N` since it matches what all the `Polygon` docstrings use, but I agree that `nverts` is easier to read if you're skimming the code only.